great_tables/_utils_render_html.py

```diff
@@ -180,7 +191,7 @@ def create_columns_component_h(data: GTData) -> str:
                     colspan=1,
                     style=_flatten_styles(styles_column_labels + styles_i),
                     scope="col",
-                    id=_process_text_id(info.column_label),
+                    id=_create_element_id(table_id, info.column_label),
```

Comment:
We can easily have duplicate column labels (e.g., using `cols_label()`) but it's much less likely to have duplicate column names. So I suggest using the column name instead of the label.   

```suggestion
                    id=_create_element_id(table_id, info.var),
```